src/main/java/org/kairosdb/core/datastore/QueryQueuingManager.java

```diff
@@ -113,6 +114,39 @@ private boolean acquireSemaphore(String queryHash, QueryMetric metric) throws In
 			return true;
 	}
 
+	public ArrayList<Pair<String, QueryMetric>> getRunningQueries()
+	{
+		ArrayList<Pair<String, QueryMetric>> runningQueriesList = new ArrayList<Pair<String, QueryMetric>>();
+		lock.lock();
+		ArrayList<String> queryKeys = new ArrayList<String>();
+		queryKeys.addAll(runningQueries.keySet());
+		try
+		{
+			for (String key : queryKeys)
```

Comment:
Just put runningQueries.keySet() here in the for loop.  No need for the extra queryKeys array list that you copy it into.

Reply:
Yeah, I thought about doing that at first but decided against it... I've gone ahead and made this change now.